### PR TITLE
el8toel9: Warn about the NVIDIA driver before upgrade

### PR DESCRIPTION
**Pull request by nielsdg**
See also https://bugzilla.redhat.com/show_bug.cgi?id=2057026

### CODE REVIEW

**Comment by nielsdg** [comment]
review please

**Review by vinzenz — CHANGES_REQUESTED** [review]
@nielsdg 
Please run `make lint_fix` to fix the import linter errors

Also please add some test for the actor:
An inspiration for your use case: https://github.com/oamg/leapp-repository/blob/00aa46919b911f93cdb4eeddf90085eda875839e/repos/system_upgrade/el7toel8/actors/checkkdegnome/tests/unit_test_checkkdegnome.py

File: repos/system_upgrade/el8toel9/actors/nvidiaproprietarydriver/actor.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    name = 'check_nvidia_proprietary_driver'
+    consumes = (ActiveKernelModulesFacts,)
+    produces = (Report,)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+
+        for fact in self.consume(ActiveKernelModulesFacts):
+            nvidia_driver_loaded = any("nvidia" in active_mod.filename for active_mod in fact.kernel_modules)
```
Comment (vinzenz) [inline repos/system_upgrade/el8toel9/actors/nvidiaproprietarydriver/actor.py:25]:
```suggestion
            nvidia_driver_loaded = any('nvidia' in active_mod.filename for active_mod in fact.kernel_modules)
```

Comment (nielsdg) [reply repos/system_upgrade/el8toel9/actors/nvidiaproprietarydriver/actor.py]: Thanks, fixed!

File: repos/system_upgrade/el8toel9/actors/nvidiaproprietarydriver/tests/test_nvidiadriver.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    current_actor_context.run()
+    assert not current_actor_context.consume(Report)
+
+
+def test_actor_with_nouveau_driver(current_actor_context):
+    without_nvidia = [
+        ActiveKernelModule(filename='nouveau', parameters=[]),
+        ActiveKernelModule(filename='kvm', parameters=[])]
+
+    current_actor_context.feed(create_modulesfacts(kernel_modules=without_nvidia))
```
Comment (vinzenz) [inline repos/system_upgrade/el8toel9/actors/nvidiaproprietarydriver/tests/test_nvidiadriver.py:32]:
```suggestion
    current_actor_context.feed(ActiveKernelModulesFacts(kernel_modules=without_nvidia))
```

Comment (nielsdg) [reply repos/system_upgrade/el8toel9/actors/nvidiaproprietarydriver/tests/test_nvidiadriver.py]: right, thanks!

File: repos/system_upgrade/el8toel9/actors/nvidiaproprietarydriver/tests/test_nvidiadriver.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+from leapp.models import ActiveKernelModule, ActiveKernelModulesFacts
+from leapp.reporting import Report
+from leapp.snactor.fixture import current_actor_context
```
Comment (vinzenz) [inline repos/system_upgrade/el8toel9/actors/nvidiaproprietarydriver/tests/test_nvidiadriver.py:3]:
```suggestion

```

Comment (vinzenz) [reply repos/system_upgrade/el8toel9/actors/nvidiaproprietarydriver/tests/test_nvidiadriver.py:3]: This is not necessary, and actually unsupported

File: repos/system_upgrade/el8toel9/actors/nvidiaproprietarydriver/actor.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    def process(self):
+
+        for fact in self.consume(ActiveKernelModulesFacts):
+            nvidia_driver_loaded = any('nvidia' in active_mod.filename for active_mod in fact.kernel_modules)
+            if nvidia_driver_loaded:
+                create_report([
+                    reporting.Title('Proprietary NVIDIA driver detected'),
+                    reporting.Summary(
+                            'Leapp has detected that the NVIDIA proprietary driver has been loaded, which also means '
+                            'the nouveau drier is blacklisted. If you upgrade now, you will end up without a '
```
Comment (Rezney) [inline repos/system_upgrade/el8toel9/actors/nvidiaproprietarydriver/actor.py:31]:
```suggestion
                            'the nouveau driver is blacklisted. If you upgrade now, you will end up without a '
```

Comment (Rezney) [reply repos/system_upgrade/el8toel9/actors/nvidiaproprietarydriver/actor.py:31]: @nielsdg Please fix this typo and we should be good to merge it. Thanks.

**Review by vinzenz — APPROVED** [review]

**Pull request merged**